### ISSUE TITLE
Create coding style guide for development of APG Examples

### ISSUE THREAD

**Issue by ZoeBijl** [w3c/aria-practices#12]
_From @MichielBijl on October 12, 2015 18:12_

We need to decide on a couple of rules for our code examples. From today's meeting:
- indentation should be four spaces;
- use semi-colons;
- use all caps for constants.

_Copied from original issue: w3c/aria#95_

**Comment by MichielBijl** [comment]
_From @LJWatson on October 12, 2015 18:15_

Suggest we use {} consistently too.

**Comment by ckundo** [comment]
I'd like to chip in on this. https://github.com/jshint/jshint/ is a good tool for enforcing conventions, and is highly configurable.

**Comment by mcking65** [comment]
Draft by Ian is at:
https://ianpouncey.github.io/code-guide/

**Comment by IanPouncey** [comment]
Add a coding style to not use background images for meaningful content - either use ::before and content or real  elements to avoid problems with Windows High Contrast Mode and content not available to a11y API.

**Comment by IanPouncey** [comment]
Add guide on JavaScript encapsulation.

**Comment by mcking65** [comment]
Based on [October 3, 2016 meeting discussion](https://www.w3.org/2016/10/03-aria-apg-minutes.html), 
@IanPouncey will create a PR that puts the guide into a /docs subdirectory in the repository. 

Ian is also updating the guide to include information about the jscsc requirements enforced by travis-ci.

**Comment by mcking65** [comment]
@IanPouncey,

Just found something else we need to call out to help people writing js for our repo ...

To avoid a sort of nasty little jscs issue, It is important that in a switch, the break is aligned with case block and not the case expression, like:

```
case expression:
  statement;
  break;
```

See [jscs issue 105](https://github.com/jscs-dev/gulp-jscs/issues/105) for an explanation. 
Tl;dr: jscs uses the format of the first break it encounters to establish the indent pattern. You could introduce hundreds of errors if you introduce new code with a differently formatted break and it just happens to be tested first.

**Comment by mcking65** [comment]
OK, @MichielBijl, @Jnurthen, I converted the content from Ian's repo, which was Jekyll and HTML into markdown and put it into the wiki. 

I don't think the Jekyll content is going to work in the gh-pages branch of our repo. And, I would really prefer to get all the project's meta content into the wiki. So, I am hoping 
[this conversion to a wiki code-guide page](https://github.com/w3c/aria-practices/wiki/Code-Guide) looks OK.

Note that I have not revised any of the words yet. I just converted and want people to tell me if it is or is not terribly hard on the eyes. I am not super familiar with styling options in the wiki, but imagine there are some ways to pretty it if it is basically functional.
